qgis-app/styles/models.py

```diff
@@ -0,0 +1,125 @@
+from django.db import models
+from django.core.validators import FileExtensionValidator
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.utils.translation import ugettext_lazy as _
+
+STYLES_STORAGE_PATH = getattr(settings,
+                                 'PLUGINS_STORAGE_PATH', 'styles/%Y')
+
+
+class StyleType(models.Model):
+    """
+    Style Type model
+    """
+
+    # symbol, name and desc
+    # symbol_type e.g. "line"
+    # name e.g. "Line"
+    symbol_type = models.CharField(
+        _('Symbol type'),
+        help_text=_('This Symbol will be used to identify the type of a style '
+                    'based on an attribute value in the uploaded XML file.'),
```

Comment:
'the value will be extracted automatically from the uploaded QGIS style XML file.'